barretenberg/cpp/src/barretenberg/execution_trace/execution_trace.hpp

```diff
@@ -24,14 +24,40 @@ template <class Flavor> class ExecutionTrace_ {
         uint32_t ram_rom_offset = 0;    // offset of the RAM/ROM block in the execution trace
         uint32_t pub_inputs_offset = 0; // offset of the public inputs block in the execution trace
 
-        TraceData(size_t dyadic_circuit_size, Builder& builder)
+        TraceData(Builder& builder, ProvingKey& proving_key)
         {
-            // Initializate the wire and selector polynomials
-            for (auto& wire : wires) {
-                wire = Polynomial(dyadic_circuit_size);
-            }
-            for (auto& selector : selectors) {
-                selector = Polynomial(dyadic_circuit_size);
+            ZoneScopedN("TraceData construction");
+            if constexpr (IsHonkFlavor<Flavor>) {
```

Comment:
ugly but seems to be necessary

Reply:
yeah don't immediately see a better way. This might all change anyway since we may not want to pre-allocate space for all of our polynomials ahead of time, which is the only reason we need to do this sharing here.